reframe/core/runtime.py

```diff
@@ -267,6 +267,113 @@ def is_env_loaded(environ):
                 for k, v in environ.variables.items()))
 
 
+def _is_valid_part(part, valid_systems):
+    for spec in valid_systems:
+        if spec[0] not in ('+', '-', '%'):
+            # This is the standard case
+            sysname, partname = part.fullname.split(':')
+            valid_matches = ['*', '*:*', sysname, f'{sysname}:*',
+                             f'*:{partname}', f'{part.fullname}']
+            if spec in valid_matches:
+                return True
+        else:
+            plus_feats = []
+            minus_feats = []
+            props = {}
+            for subspec in spec.split(' '):
+                if subspec.startswith('+'):
+                    plus_feats.append(subspec[1:])
+                elif subspec.startswith('-'):
+                    minus_feats.append(subspec[1:])
+                elif subspec.startswith('%'):
+                    key, val = subspec[1:].split('=')
+                    props[key] = val
+
+            have_plus_feats = all(
```

Comment:
This is just a question, in case on no `plus_feats` do we consider `have_plus_feats` to be True?

Reply:
Good question, but yes we need to consider it as true due to the check below:

```python
            if have_plus_feats and not have_minus_feats and have_props:
                return True
```

One of the three will have at least an item to check, so if we didn't assume `True` for empty `plus_feats`, this test would fail. Similarly the default `False` for the `any()` of the `minus_feats` serves the same purpose.